pkg/assetfetcher/http.go

```diff
@@ -154,19 +157,19 @@ func (c *NexposeAssetFetcher) makeRequest(ctx context.Context, wg *sync.WaitGrou
 }
 
 // newNexposeSiteAssetsRequest builds URL we'll use to make the request to Nexpose's site assets endpoint
-func newNexposeSiteAssetsRequest(host string, siteID string, page int, size int) (*http.Request, error) {
-	u, err := url.Parse(host)
+func (c *NexposeAssetFetcher) newNexposeSiteAssetsRequest(siteID string, page int) (*http.Request, error) {
+	u, err := url.Parse(c.Host)
```

Comment:
Suggestion: Now that this value is being loaded through a settings component you have the option of parsing the URL in the component's `New()` method. This would allow you to return an error early in the startup process rather than hitting that same error here.

If you did this you would also be able to inject a `*url.URL` which would prevent you from needing to parse in this part of the code at all.

Reply:
got it. I'll make that change. Thanks!